index.js

```diff
@@ -1 +1,169 @@
-// implement your API here
+const express = require("express");
+const db = require("./data/db");
+
+const server = express();
+const port = 3000;
+
+server.use(express.json());
+
+/**
+|--------------------------------------------------
+|  GET    | /api/users     | Returns an array of all the user objects contained in the database.
+|--------------------------------------------------
+*/
+
+server.get("/api/users", (request, response) => {
+  db.find()
+    .then(users => {
+      response.status(200).json(users);
+    })
+    .catch(err => {
+      response.status(500).json({
+        error: "The users information could not be retrieved."
+      });
+    });
+});
+
+/**
+|--------------------------------------------------
+|  POST   | /api/users     | Creates a user using the information sent inside the `request body`.
+|--------------------------------------------------
+
+**/
+
+server.post("/api/users", (request, response) => {
+  const { body } = request;
+  const { name, bio } = body;
+
+  if (!name || !bio) {
+    response.status(400).json({
+      errorMessage: "Please provide name and bio for the user."
+    });
+  } else {
+    db.insert(body)
+      .then(user => {
+        response.status(201).json({
+          user
```

Comment:
you may not necessarily need this to be encapsulated in an object. It's already an object.